network/src/protocols/health_checker/builder.rs

```diff
@@ -46,44 +23,21 @@ impl HealthCheckerBuilder {
         network_tx: HealthCheckerNetworkSender,
         network_rx: HealthCheckerNetworkEvents,
     ) -> Self {
-        HealthCheckerBuilder::new(HealthCheckerBuilderConfig {
+        let service = HealthChecker::new(
             network_context,
             time_service,
-            ping_interval_ms,
-            ping_timeout_ms,
-            ping_failures_tolerated,
             network_tx,
             network_rx,
-        })
-    }
-
-    pub fn build(&mut self, executor: &Handle) -> &mut Self {
-        // Can only build once;  must build before starting.
-        assert!(!self.built);
-        assert!(!self.started);
-        self.built = true;
-        if let Some(config) = self.config.take() {
-            let _guard = executor.enter();
```

Comment:
You may or may not have an issue with this line since there isn't an equivalent elsewhere.